internal/mode/static/state/graph/tlsroute_test.go

```diff
@@ -440,16 +483,68 @@ func TestBuildTLSRoute(t *testing.T) {
 					"Service configured with IPv4 family but NginxProxy is configured with IPv6",
 				)},
 				Attachable: true,
+				Valid:      true,
 			},
 			gatewayNsNames: []types.NamespacedName{gatewayNsName},
 			services: map[types.NamespacedName]*apiv1.Service{
 				svcNsName: ipv4Svc,
 			},
-			name: "service and npcfg ip family mismatch",
 			npCfg: NginxProxy{
 				Source: &ngfAPI.NginxProxy{Spec: ngfAPI.NginxProxySpec{IPFamily: helpers.GetPointer(ngfAPI.IPv6)}},
 				Valid:  true,
 			},
+			resolver: alwaysTrueRefGrantResolver,
+			name:     "service and npcfg ip family mismatch",
+		},
+		{
+			gtr: diffNsBackendRef,
```

Comment:
How does this test case differ than the 

`"valid; same namespace"` one below?

Reply:
The backend ref is in a different namespace which means the ref grant resolver must return true in order for the backend ref to be valid.

Reply:
It's in a different namespace than which resource?

Reply:
Than the route

Reply:
So the route is defined as `gtr: diffNsBackendRef,` which is in the `diff` namespace, isn't the BackendRef also in the `SvcNsName:   diffSvcNsName,` which is the `diff` namespace?

Or am I missing something with the logic?

Reply:
Every TLSRoute is created in the `test` namespace:

```
func createTLSRoute(
	hostname gatewayv1.Hostname,
	rules []v1alpha2.TLSRouteRule,
	parentRefs []gatewayv1.ParentReference,
) *v1alpha2.TLSRoute {
	return &v1alpha2.TLSRoute{
		ObjectMeta: metav1.ObjectMeta{
			Namespace: "test",
			Name:      "tr",
		},
		Spec: v1alpha2.TLSRouteSpec{
			CommonRouteSpec: gatewayv1.CommonRouteSpec{
				ParentRefs: parentRefs,
			},
			Hostnames: []gatewayv1.Hostname{hostname},
			Rules:     rules,
		},
	}
}
```

the `diffNsBackendRef` variable creates a TLSRoute using the above function and passes the rules which contains a backend ref in the `diff` namespace:

```
	diffNsBackendRef := createTLSRoute("app.example.com",
		[]v1alpha2.TLSRouteRule{
			{
				BackendRefs: []gatewayv1.BackendRef{
					{
						BackendObjectReference: gatewayv1.BackendObjectReference{
							Name:      "hi",
							Port:      helpers.GetPointer[gatewayv1.PortNumber](80),
							Namespace: helpers.GetPointer[gatewayv1.Namespace]("diff"),
						},
					},
				},
			},
		},
		[]gatewayv1.ParentReference{
			parentRef,
		},
	)
```

Reply:
Ahhh okay this is making a lot more sense, thanks!

Reply:
np 🙂